### DISCUSSION TITLE
How to investigated > Error: Worker exited unexpectedly

### DISCUSSION THREAD

**Discussion by JesusTheHun** [tinylibs/tinypool#87] [Q&A]
Hi,

I'm using vitest and I'm facing this error a lot, `Error: Worker exited unexpectedly`.

I'm doing some pretty heavy dark magic: `vitest -> tinypool -> fork -> async_hook -> mock -> proxy -> native code`. 
So I would like to make a deep dive to find out what's going on, but the only trace I have is : 

> Error: Worker exited unexpectedly
 ❯ ChildProcess.<anonymous> node_modules/tinypool/dist/esm/index.js:185:34
 ❯ ChildProcess.emit node:events:525:35
 ❯ ChildProcess._handle.onexit node:internal/child_process:291:12

Also, while it happens very often (roughly 4/5), it does not happen all the time.
Does one know the best way to get to the bottom of such errors ?

**Comment by AriPerkkio** [comment]
Are you mocking any Node's built-in modules like `node:fs` or `node:child_process`? Or calling `process.exit()` from test?

> Does one know the best way to get to the bottom of such errors ?

Isolate the error into a minimal reproduction first.

**Reply by JesusTheHun** [reply]
I'm not mocking any built-in module.

Isolate the error of course, but since there is no stack trace it's very time consuming. I was hoping for a way to reduce the scope of my investigations.

**Comment by akanshaaa19** [comment]
Hey does anyone know how to solve this?

**Reply by AriPerkkio** [reply]
No one knows how to reproduce this.
